src/ecalc/libraries/libecalc/common/libecalc/core/graph_result.py

```diff
@@ -230,30 +238,67 @@ def get_asset_result(self) -> libecalc.dto.result.EcalcModelResult:
-                        "parent": self.graph.get_predecessor(consumer_id),
-                        "component_level": consumer_node_info.component_level,
-                        "componentType": consumer_node_info.component_type,
-                        "emissions": self._parse_emissions(self.emission_results[consumer_id])
-                        if consumer_id in self.emission_results
-                        else [],
-                        "energy_usage_cumulative": consumer_result.component_result.energy_usage.to_volumes()
-                        .cumulative()
-                        .dict(),
-                        "power_cumulative": consumer_result.component_result.power.to_volumes()
-                        .to_unit(Unit.GIGA_WATT_HOURS)
-                        .cumulative()
-                        .dict()
-                        if consumer_result.component_result.power is not None
-                        else None,
-                    },
+            if self.graph.get_component(consumer_id).component_type == ComponentType.COMPRESSOR:
+                component = self.graph.get_component(consumer_id)
+
+                requested_inlet_pressure = self.get_pressures_from_temporal_models(
+                    component.energy_usage_model, component.regularity, CompressorInputPressures.inlet_pressure
+                )
+                requested_outlet_pressure = self.get_pressures_from_temporal_models(
+                    component.energy_usage_model, component.regularity, CompressorInputPressures.outlet_pressure
+                )
+
+                sub_components.append(
+                    parse_obj_as(
+                        libecalc.dto.result.ComponentResult,
+                        {
+                            **consumer_result.component_result.dict(),
+                            "name": consumer_node_info.name,
+                            "parent": self.graph.get_predecessor(consumer_id),
+                            "component_level": consumer_node_info.component_level,
+                            "componentType": consumer_node_info.component_type,
+                            "emissions": self._parse_emissions(self.emission_results[consumer_id])
+                            if consumer_id in self.emission_results
+                            else [],
+                            "energy_usage_cumulative": consumer_result.component_result.energy_usage.to_volumes()
+                            .cumulative()
+                            .dict(),
+                            "power_cumulative": consumer_result.component_result.power.to_volumes()
+                            .to_unit(Unit.GIGA_WATT_HOURS)
+                            .cumulative()
+                            .dict()
+                            if consumer_result.component_result.power is not None
+                            else None,
+                            "requested_inlet_pressure": requested_inlet_pressure,
```

Comment:
a lot of duplicated code. possible to add "custom properties" as a separate dict depending on type? what if we get custom properties for pumps?